### PR TITLE
Issue #53 updates to contribution guidelines

### PR DESCRIPTION
**Pull request by mstoodle**
In addition to the relatively simple commits outlined in the commit messages, I also restructured the document a bit to move the commit and pull request suggestions closer to the top.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -20,6 +18,33 @@ languages to enrich the options available to programmers.
...
+## GitHub Issues
+
+This project uses GitHub Issues to track ongoing development and issues. Be sure
+to search for existing bugs before you create another one. Contributions are always welcome!
+
+- [https://github.com/eclipse/omr/issues](https://github.com/eclipse/omr/issues)
+
+## Submitting a contribution
+You can propose contributions by sending pull requests through GitHub. Following these guidelines
+will help us to merge your pull requests smoothly:
```
Comment (alin00) [inline CONTRIBUTING.md]: The markdown formatting isn't quite right - it's messed up when I preview the file.  I think you might need to add blank lines between paragraphs, and maybe outdent the commit dtemplate.

**Comment by alin00** [comment]
Need to reference the issue number in a comment: This pull request is for issue #53.

**Comment by alin00** [comment]
@charliegracie wasn't quite satisfied with the commit template I proposed in #53. Do you have anything to add here?

**Comment by charliegracie** [comment]
I would like the commit comment to be something similar to my recent one. Things in wrapped in "()" are only required if there is an issue

\(Fix \#32\) Summary up to approximately 50 characters
\<blank line\>
Detailed information
\<blank line\>
\(Issue: \#\)
Signed-off: Name \<email\>

**Comment by alin00** [comment]
> (Fix #32) Summary up to approximately 50 characters

I slightly prefer not to use github's autoclose feature, so I wouldn't use "Fix" here. But it's not critical to me.

Based on prior experience, people will get creative about the case and punctuation of this, so any tooling we have needs to tolerate lots of variants of "FIX #nnn", "Fix # nnn" etc.

File: CONTRIBUTING.md (hunk 1 above)
Comment (mstoodle) [inline CONTRIBUTING.md]: yeah, one missing blank line :( ... thanks

**Comment by mstoodle** [comment]
Ok, i'll make these updates.

@charliegracie To confirm: you do not want the issue number on the log output until the issue is fixed?

I have a slight worry that we're descending somewhat close to a degree of pedantry / rigidity that may discourage some external contributors, but hopefully we haven't crossed that threshold yet.

**Comment by alin00** [comment]
>  the first "()" should only be used when the issue is completely resolved.
...
> To confirm: you do not want the issue number on the log output until the issue is fixed?

I really think we should have the issue number in every commit associated with the issue. Insisting on putting it only in the last commit is tricky. I often think a commit is the last, and then it isn't.

**Comment by mstoodle** [comment]
Agree with Angela. That was why I asked for clarification to make sure I understood the suggestion...

**Comment by charliegracie** [comment]
No I agree that every commit should have a reference to the issue. I was only stating that we only use one of the auto close tags in the final commit. Thinking about it more though it doesn't matter since all commits will be delivered when the PR gets merged. 

I think the Issue on the first line just as a reference takes too much real estate if it is not serving an automated process. That is just my preference and I am ok with proposal.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -28,22 +28,27 @@ to search for existing bugs before you create another one. Contributions are alw
...
+    Short, simple summary that must include #n
+
     Detailed information if required
 
     Signed-off-by: First Last <email>
     ```
+
 5. Ensure that "make test" passes all tests before you submit a Pull Request.
-6. Reference the Issue number when you create the Pull Request comment.
+6. The pull request comment should reference the issue number in its title, just as in the commit message template.
```
Comment (alin00) [inline CONTRIBUTING.md]: For the github cross-referencing to work, the issue number has to be in a discussion comment of the pull request, and not just the pull request title.  Almost everyone already puts the issue number in the pull request title now, and we get no cross-ref.

**Comment by alin00** [comment]
Thanks.  It would have been nice to hyperlink to [CodingStandard.md](https://github.com/eclipse/omr/blob/master/doc/CodingStandard.md), but I'll merge this.

**Pull request merged**